### DISCUSSION TITLE
Is compilation to webgl1 supported?

### DISCUSSION THREAD

**Discussion by leerle** [gfx-rs/wgpu#4026] [Q&A]
Some users have older computers, and I hope to compile to webgl1. At present, WGPU supports compiling products into WebGL1?

**Comment by cwfitzgerald** [comment]
We do not support WebGL1 (and by extention, GLES 2). This is in part because WebGL1 uses a completely different binding and upload model, which even if we were to support it, would likely require the user would need to write different wgpu code. 

If you want to support WebGL1, I would recommend either using something like `glow` directly, or using wgpu for your webgl2/native path and have a separate backend for webgl1.

I do note though, WebGL2 is based on GLES 3.0, which is targetting phones from 2012. I would be quite surprised if you find a ton of devices that don't support WebGL2.
